### PR TITLE
Update format and code in doc

### PR DESCRIPTION
**Pull request by xuebinsu**
- Add :code: directives for highlighting.
- Update code in notebooks to use latest features
- Update descriptions in doc to reflect concept changes
- Re-run notebooks to make sure code are correct.

### CODE REVIEW

File: doc/source/notebooks/pandas.ipynb [hunk 1]
```diff
@@ -1,2066 +1,2071 @@
...
-        },
-        "vscode": {
-            "interpreter": {
-                "hash": "aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49"
-            }
-        }
-    },
-    "nbformat": 4,
-    "nbformat_minor": 4
+ "cells": [
```
Comment (beeender) [inline doc/source/notebooks/pandas.ipynb:2]: Did this file actually change or only formatting? I would just skip this one if so.

**Comment by Quasimewdo** [comment]
The last cell of `abalone.ipynb` generates an error

File: doc/source/notebooks/basic.ipynb [hunk 2]
```diff
@@ -103,9 +100,9 @@
             "cell_type": "markdown",
             "metadata": {},
             "source": [
-                "### Selecting the Database of Your Data\n",
+                "### Getting Access to Database\n",
                 "\n",
-                "To begin with, we need to select the database that contains the data we want:"
+                "To get access to the database we want:"
```
Comment (yihong0618) [inline doc/source/notebooks/basic.ipynb:105]: Access to the database is OK I think.

File: tests/test_func.py [hunk 3]
```diff
@@ -691,6 +691,23 @@ def sum_count(result: sum_count_type, val: int) -> sum_count_type:
...
+    @gp.create_function
+    def test_plpy() -> bool:
+        # Function with name starting with "__func_" will only be created by
+        # exec() when using source serialization, i.e. when dill is not
+        # available on server.
+        #
+        # If dill is used for deserialization, plpy will not be available.
+        return "plpy" in globals().keys() or (
+            len([True for name in globals().keys() if name.startswith("__func_")]) == 0
+        )
```
Comment (yihong0618) [inline tests/test_func.py:704]: len([True for name in globals().keys() if name.startswith("__func_")]) == 0 -> all([name.startswith("__func_") for name in globals().keys()]) ?

File: tests/test_func.py [hunk 4]
```diff
@@ -738,12 +753,10 @@ class Student(SimpleNamespace):
 
     @gp.create_function
     def student(name: str, age: int) -> Student:
-        # Fallback when the pickle lib "dill" cannot be used on server
-        # so that the case can always pass.
-        try:
+        # Fallback when the pickle lib "dill" is not available on server.
+        if len([True for name in globals().keys() if name.startswith("__func_")]) == 0:
```
Comment (yihong0618) [inline tests/test_func.py:757]: or make this a function?

File: doc/source/notebooks/pandas.ipynb [hunk 5]
```diff
@@ -1,2066 +1,2071 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Selecting Rows by Predicates\n",
+    "\n",
+    "The `[]` operator can also be used to select a subset of rows, a.k.a filtering.\n",
+    "\n",
+    "Say we want the information of student named \"alice\", with pandas we can do"
```
Comment (beeender) [inline doc/source/notebooks/pandas.ipynb:737]: `:`

File: doc/source/notebooks/pandas.ipynb [hunk 6]
```diff
@@ -1,2066 +1,2071 @@
...
+    "\n",
+    "Like pandas, GreenplumPython provides powerful building blocks to make transformation easier.\n",
+    "\n",
+    "### Data Ordering\n",
+    "\n",
+    "Having the data sorted in a desired order makes it convenient for many analytical tasks, such as statistics.\n",
+    "\n",
+    "pandas supports sorting the data (a.k.a values) of a `DataFrame` by columns.\n",
+    "\n",
+    "For example, we can sort in pandas the dataframe of student info by \"age\" and then \"name\", both in descending order with"
```
Comment (beeender) [inline doc/source/notebooks/pandas.ipynb:999]: `:`

File: doc/source/notebooks/pandas.ipynb [hunk 7]
```diff
@@ -1,2066 +1,2071 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Data Grouping\n",
+    "\n",
+    "Like ordering, Grouping data based on distinct set of values of columns can also facilitate analytical tasks.\n",
+    "\n",
+    "Data grouping is often associated with aggregate functions to obtain data summaries. \n",
+    "\n",
+    "For example, suppose we want to the number of students of different age. In pandas, we can do"
```
Comment (beeender) [inline doc/source/notebooks/pandas.ipynb:1533]: For example, suppose we want to get the students' count of different ages. In pandas, we can do:

File: doc/source/notebooks/pandas.ipynb [hunk 8]
```diff
@@ -1,2066 +1,2071 @@
...
+    "import numpy as np\n",
+    "\n",
+    "pd_df.groupby(\"age\").apply(lambda df: np.count_nonzero(df[\"name\"]))\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In GreenplumPython, what we need to do is"
```
Comment (beeender) [inline doc/source/notebooks/pandas.ipynb:1565]: `:`

File: doc/source/notebooks/pandas.ipynb [hunk 9]
```diff
@@ -1,2066 +1,2071 @@
...
+   "source": [
+    "### Data Deduplication\n",
+    "\n",
+    "Data deduplication is to return a new data structure containing only the distinct set of values in the selected columns.\n",
+    "\n",
+    "This operation is well supported in GreenplumPython, also in a very similar way to pandas.\n",
+    "\n",
+    "For example, suppose we want to draw a representative sample containing students for each distinct age,\n",
+    "\n",
+    "With pandas, we can do"
```
Comment (beeender) [inline doc/source/notebooks/pandas.ipynb:1624]: `:`

File: doc/source/notebooks/pandas.ipynb [hunk 10]
```diff
@@ -1,2066 +1,2071 @@
...
+    "student.apply(lambda t: count.distinct(t[\"age\"]))\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Joins\n",
+    "\n",
+    "Joins are operations that combines two data structures horizontally in a sensible way.\n",
```
Comment (beeender) [inline doc/source/notebooks/pandas.ipynb:1786]: combines -> combine

File: doc/source/notebooks/pandas.ipynb [hunk 11]
```diff
@@ -1,2066 +1,2071 @@
...
+   ],
+   "source": [
+    "pd_df.merge(pd_df, on=\"age\", suffixes=(\"\", \"_2\"))\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Similarily, in GreenplumPython, we can do"
```
Comment (beeender) [inline doc/source/notebooks/pandas.ipynb:1883]: `:`

**Review by yihong0618 — APPROVED** [review]
LGTM, cool

File: greenplumpython/dataframe.py [hunk 12]
```diff
@@ -1,22 +1,22 @@
 """
-`DataFrame` is the core data structure in GreenplumPython.
+DataFrame is the core data structure in GreenplumPython.
 
-Conceptually, a `DataFrame` is a two-dimensional structure containing data.
+Conceptually, a DataFrame is a two-dimensional structure containing data.
 
-In the data science world, a :class:`DataFrame` in GreenplumPython is similar to a `DataFrame
-<https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`_ in `pandas <https://pandas.pydata.org/>`_,
+In the data science world, a DataFrame in GreenplumPython, refered to as :code:`gp.DataFrame`, is similar to a
```
Comment (Quasimewdo) [inline greenplumpython/dataframe.py:6]: referred

File: greenplumpython/dataframe.py [hunk 13]
```diff
@@ -898,12 +897,12 @@ def save_as(
         :func:`~db.Database.create_dataframe(table_name)` to create a new :class:`~dataframe.Dataframe` next time.
 
         Args:
-            table_name : str
-            temp : bool : if table is temporary
-            column_names : List : list of column names
-            storage_params: dict: storage_parameter of gpdb, reference
+            table_name: name of table in database, requied to be unique in the schema.
```
Comment (Quasimewdo) [inline greenplumpython/dataframe.py:900]: required

**Review by Quasimewdo — APPROVED** [review]
LGTM

**Pull request merged**